src/common/utils/url.ts

```diff
@@ -204,6 +204,43 @@ function isCustomLoginURL(url: URL | string, server: ServerFromURL, teams: TeamW
     return false;
 }
 
+function isTrustedPluginRoute(serverUrl: URL | string, inputUrl: URL | string): boolean {
+    if (!serverUrl || !inputUrl) {
+        return false;
+    }
+
+    const parsedURL = parseURL(inputUrl);
+    const server = getServerInfo(serverUrl);
+    if (!parsedURL || !server || (!equalUrlsIgnoringSubpath(server.url, parsedURL))) {
+        return false;
+    }
+
+    const formattedPathName = parsedURL.pathname.toLowerCase();
+
+    if (!formattedPathName.startsWith(`${server.subpath}plugins/`) || !formattedPathName.startsWith('/plugins/')) {
+        return false;
+    }
```

Comment:
I think it would make sense to just check for `isPluginRoute` and then checking to see if it's the specific path we have below. I would add those paths into an array of some kind and check if it's one of those special cases.

Reply:
The thing is that we still need to get the server info, parse the URL and check the prefix in order to clean the pathname to make sure we check against a well-defined path. So it would end up like this:

```ts
    if (!isPluginUrl(serverUrl, inputUrl)) {
        return false;
    }

    // We still need to get the server and parse the URL, which isPluginUrl already did
    const server = getServerInfo(serverUrl);
    const parsedUrl = parseURL(inputUrl);

    // We also need to check that they're not undefined for typescript
    // to not complaint, even if isPluginUrl already checked
    if (!server || !parsedUrl) {
        return false;
    }

    const formattedPathName = parsedUrl.pathname.toLowerCase();
    
    // These checks were also done already
    let cleanPathname;
    if (formattedPathName.startsWith(`${server.subpath}plugins/`)) {
        cleanPathname = formattedPathName.replace(`${server.subpath}plugins/`, '');
    } else 
        // this branch now relies on the actual implementation of isPluginUrl.
        // we could alternatively still check for startsWith('/plugins') in an else if branch
        // and return false in the final else
        formattedPathName.replace('/plugins/', '');
    }

    const allowedRoutes = [
        'com.mattermost.plugin-channel-export/api/v1/export',
    ];

    return allowedRoutes.includes(cleanPathname);
```

This is more or less the same we had before, but now we have code running twice (check the comments above) and rely on the inner implementation of `isPluginUrl`.

Maybe there is some other utility function already written for making this easier? If not, I rather keep the actual implementation.

Thoughts?

Reply:
If the route is always going to be of the form `/plugins/com.mattermost.plugin-channel-export/api/v1/export`, maybe instead we try to keep it simple for now and just write another helper that just calls `isUrlType` instead of first checking if it's a plugin url and THEN checking if its the exception.

```suggestion
function isChannelExportUrl(serverUrl: URL | string, inputURL: URL | string) {
    return isUrlType('plugins/com.mattermost.plugin-channel-export/api/v1/export', serverUrl, inputURL);
}
```

Reply:
Yeah, that makes more sense. Even my original proposal was premature generalization :) Let me test that and push it.
 Thanks for the discussion!